packages/logger/tests/unit/logBuffer.test.ts

```diff
@@ -107,4 +107,34 @@ describe('flushBuffer', () => {
     expect(console.debug).toHaveBeenCalledTimes(0);
     expect(console.warn).toHaveBeenCalledTimes(1);
   });
+
+  it('outputs a warning when buffered logs have been evicted', () => {
+    // Prepare
+    const logger = new TestLogger({ logLevel: 'ERROR' });
+    logger.enableBuffering();
+    logger.setbufferLevelThreshold(LogLevelThreshold.INFO);
+
+    // Act
+    const longMessage = 'blah'.repeat(10);
+
+    let i = 0;
+    while (i < 4) {
+      logger.info(
+        `${i} This is a really long log message intended to exceed the buffer ${longMessage}`
+      );
+      i++;
+    }
```

Comment:
This was a bit fiddly to configure correctly so that we'd overflow the buffer. If we change the default size of the buffer in future this might break.

Reply:
Agreed, but there's another issue open in which we'll make the buffer size configurable, so once that's done we can modify this test to set a fixed one.